### PR TITLE
feat!: release management, docker build

### PR DESCRIPTION
**Pull request by eshepelyuk**
* milti stage docker build
* docker image and helm chart published only on release
* release happens only on git tag
* tag format doesn't include v prefix
* helm chart version synchornized with docker version
* version is taken only from git tag

Signed-off-by: Ievgenii Shepeliuk <eshepelyuk@gmail.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,8 +1,23 @@
+FROM golang:1.12-alpine as build
```
Comment (srenatus) [inline Dockerfile:1]: [nit] we should bump this version (if we can, it looks like the code hasn't seem much love lately)

**Review by srenatus — APPROVED** [review]
The change in isolation looks good to me; but I'm not really proficient with helm, let alone release management of hemp charts. So, if this is the right thing? -- I wouldn't be able to tell 😅

**Review by anderseknert — APPROVED** [review]
LGTM. The only thing I wonder about is the removal of the makefile. The benefit of having that is that it's easy to perform the same tasks run in CI locally. But looking at what's needed it's essentially just `docker build`, so perhaps it's redundant?

**Comment by ievgenii-shepeliuk** [comment]
> LGTM. The only thing I wonder about is the removal of the makefile. The benefit of having that is that it's easy to perform the same tasks run in CI locally. But looking at what's needed it's essentially just `docker build`, so perhaps it's redundant?

Yes, it is redundant. docker build does the build locally.
Other tasks, like release and publish - should be sone only via CI.

**Comment by eshepelyuk** [comment]
> LGTM. The only thing I wonder about is the removal of the makefile. The benefit of having that is that it's easy to perform the same tasks run in CI locally. But looking at what's needed it's essentially just `docker build`, so perhaps it's redundant?

Yes, you're right. Build could be done via `docker build`, and other tasks like release and publish should be done from CI.

File: Dockerfile (hunk 1 above)
Comment (eshepelyuk) [reply Dockerfile]: Changed to 1.15. More recent versions requre additional rework - and I am not golang dev.

**Comment by eshepelyuk** [comment]
Thanks for approving this.
I plan to kick a new major release during this week.

**Pull request merged**